### ISSUE TITLE
Editor Render Settings are bogus / viewport metrics not reliably visible

### ISSUE THREAD

**Issue by JSandusky** [PanosK92/Directus3D#10]
The render switches in the editor have some sort of tangled dependency chain.

Turning on `sharpen` activates the whole pipe except FXAA, tonemap or bloom seem to be required for the viewport to even update, etc.

Most importantly, the viewport/render metrics info isn't visible without basically everything turned on.

Lastly the metrics and the profiler output have a mismatch. Although the profiler says things are swimming along nicely, it's actually only doing around 40fps (that's the entire post-stack active [remember, the checkboxes are lying, only FXAA is off] on an Intel HD4000 - not surprising), so there's so unprofiled regions laying about.

![image](https://user-images.githubusercontent.com/5519639/43671132-7732595a-9762-11e8-881d-9cd2f9927e6d.png)

---

Good job so far though, in general everything's much faster than the last time I tried it out.

**Comment by JSandusky** [comment]
This came from the AppVeyor binaries (1.0.121, @ https://github.com/PanosK92/Directus3D/commit/41bcab265011671c17918780e556b73436978308) .

**Comment by PanosK92** [comment]
Hi JSandusky,

The post effect render passes are indeed tangled.
Most of the passes require the result of the one before them, which creates a small dependency chain.
The dependency chain is handled poorly on my side and requires rework.

You are correct about the metrics as well, they report CPU time, which is misleading.
GPU timing requires me to do some D3D11 work which is long overdue as it's vital.

I am actually working on the rendering backend right now.
I'll take care of the issues you mentioned in the following commits and comment here again.

Thank you for taking the time to let me know about these issues.

**Comment by JSandusky** [comment]
I'd think wrapping `Present` in profiling would cover that, but GPU metrics are useful.

**Comment by PanosK92** [comment]
It would be an improvement, however the reading will still be unreliable.

If you were to call std::chrono::high_resolution_clock::now(), before and after a draw call, you wouldn't measure GPU time, only how long the graphics driver took to queue up that call. 

D3D11 provides ID3D11Query objects which can be used to read an internal GPU clock and should allow for accurate readings.

**Comment by PanosK92** [comment]
- GPU timings are now available in the editor's Profiler window, in a new column.
- Post-process effect stack getting tangled when tweaked via the editor, has been fixed.
- Fixed a bug that could possibly offer you a noticable performance gain.

Will still leave this issue open for some time, in order to make sure that the changes are stable.

**Comment by PanosK92** [comment]
The crash is not easy to reproduce but I got it randomly as well. The render target view of the swapchain's back buffer seems to become invalid somehow, so when it's about to be bound to the GPU you get this crash.

The viewport size doesn't seem to be the cause for me but I fixed it anyway as it was ridiculously small.
These kind of instabilities are the result of a re-worked D3D11 backend, so I have a few guesses on what changes might have caused it. Shouldn't take long to squash it.

Thanks for letting me know JSandusky!
